typed_python/StringType.cpp

```diff
@@ -814,3 +814,96 @@ void StringType::assign(instance_ptr self, instance_ptr other) {
 
     destroy((instance_ptr)&old);
 }
+
+#define max(a,b) a < b ? b : a;
+
+void StringType::join(StringType::layout **outString, StringType::layout *separator, ListOfType::layout *toJoin) {
+
+    if (!outString)
+        throw std::invalid_argument("missing return argument");
+
+    StringType::Make()->constructor((instance_ptr)outString);
+
+    // return empty string when there is nothing to join
+    if(toJoin->count == 0)
+    {
+        return;
+    }
+
+    static ListOfType *inputList = ListOfType::Make(StringType::Make());
```

Comment:
this variable should be called `listOfStrType`. OTherwise it looks like a data variable.

Reply:
done